app/models/age_requirement.rb

```diff
@@ -0,0 +1,27 @@
+class AgeRequirement
+  include Neo4j::ActiveNode
+  has_many :out, :requirements, type: :HAS_REQUIREMENT, model_class: [:OrRequirementGroup, :AndRequirementGroup]
+  property :min_age
+  property :max_age
+
+  def ineligible?(attributes)
+    return false unless attributes[:age].present?
+
+    met_requirements = requirements_present.map { |requirement| send("meets_#{requirement}_requirement?", attributes) }
```

Comment:
Personal preference - I don't really like using `send` because it obfuscates where and when methods are being called, but I can see why you've used it here. Is it optimising too soon though? 🤷‍♂

Reply:
Yeah, I'm not a huge fan. I think your suggestion is an excellent way to deal with it. The only alternative I could think of was mildly hideous nested if statements and this felt slightly better but....

Reply:
Yep, it does feel a bit "six of one, half a dozen of the other".  

I think I probably would have done something like:

```
def ineligible?(attributes)
  return false unless met_min_age_requirement?(attributes)
  return false unless met_max_age_requirement?(attributes)
  true
end

def meets_min_age_requirement?(attributes)
  return false unless read_attribute(:min_age).present?
  attributes[:age] >= min_age
end
.
.
.
```

to avoid any nested `ifs`. Trade offs though - `read_attribute` would be duplicated in each `_requirement?` method, but there's no meta-programming. Also scaling this with more criteria might be annoying. 🤷‍♂ 

It's not really a better solution though, just different.